types/node/ts4.8/http.d.ts

```diff
@@ -113,56 +114,87 @@ declare module 'http' {
+        agent?: Agent | boolean | undefined;
+        auth?: string | null | undefined;
+        // https://github.com/nodejs/node/blob/master/lib/_http_client.js#L278
+        createConnection?:
+            | ((options: ClientRequestArgs, oncreate: (err: Error, socket: Socket) => void) => Socket)
+            | undefined;
+        defaultPort?: number | string | undefined;
+        family?: number | undefined;
+        headers?: OutgoingHttpHeaders | undefined;
+        hints?: LookupOptions['hints'];
         host?: string | null | undefined;
         hostname?: string | null | undefined;
-        family?: number | undefined;
-        port?: number | string | null | undefined;
-        defaultPort?: number | string | undefined;
+        insecureHTTPParser?: boolean | undefined;
         localAddress?: string | undefined;
-        socketPath?: string | undefined;
+        localPort?: number | undefined;
+        lookup?: LookupFunction | undefined;
         /**
-         * @default 8192
+         * @default 16384
          */
         maxHeaderSize?: number | undefined;
         method?: string | undefined;
         path?: string | null | undefined;
-        headers?: OutgoingHttpHeaders | undefined;
-        auth?: string | null | undefined;
-        agent?: Agent | boolean | undefined;
-        _defaultAgent?: Agent | undefined;
-        timeout?: number | undefined;
+        port?: number | string | null | undefined;
+        protocol?: string | null | undefined;
         setHost?: boolean | undefined;
-        // https://github.com/nodejs/node/blob/master/lib/_http_client.js#L278
-        createConnection?:
-            | ((options: ClientRequestArgs, oncreate: (err: Error, socket: Socket) => void) => Socket)
-            | undefined;
-        lookup?: LookupFunction | undefined;
+        signal?: AbortSignal | undefined;
+        socketPath?: string | undefined;
+        timeout?: number | undefined;
+        uniqueHeaders?: (string | string[])[] | undefined;
     }
     interface ServerOptions<
         Request extends typeof IncomingMessage = typeof IncomingMessage,
         Response extends typeof ServerResponse = typeof ServerResponse,
     > {
+        /**
+         * Specifies the `IncomingMessage` class to be used. Useful for extending the original `IncomingMessage`.
+         */
         IncomingMessage?: Request | undefined;
+        /**
+         * Specifies the `ServerResponse` class to be used. Useful for extending the original `ServerResponse`.
+         */
         ServerResponse?: Response | undefined;
         /**
-         * Optionally overrides the value of
-         * `--max-http-header-size` for requests received by this server, i.e.
-         * the maximum length of request headers in bytes.
-         * @default 8192
+         * Sets the timeout value in milliseconds for receiving the entire request from the client.
+         * @see Server.requestTimeout for more information.
+         * @default 5000
```

Comment:
```suggestion
         * @default 300000
```
Not sure here though:
https://nodejs.org/docs/latest-v18.x/api/http.html#httpcreateserveroptions-requestlistener